### PR TITLE
Option to submit electronic files for other users on Instructor Submission page

### LINKED ISSUE #1031: Instructor / TA upload of .pdf documents (then graded similar to electronic upload by student)
To facilitate grading of scanned tests, etc.

### PR DESCRIPTION
**Pull request by wangj35**
* Closes #1031 
* Two radio buttons (visible only to instructor) to select between "Normal Submission" and "Make Submission for Student. Normal submission is selected by default, but the page uses cookies to remember the most recent selection in case the instructor is uploading many files for students in a row.
* Error message if an invalid user id is entered. 
* Team ids do not work, but you can submit for a student who is part of a team. 
* Error message if you try to submit for a student not part of a team.
* Slight changes to SubmissionControllerTester to account for user_id being passed in.
* user_assignment_settings.json now has two new fields for each object in history , "type" for either upload or select, "who" for which user was uploading or selecting the active version

### CODE REVIEW

**Comment by wangj35** [comment]
Andrew tried this branch on his computer. We have the same version of Chrome and were both in the most recent commit of my branch, but in his Chrome debugger it would say "validateStudentId is not defined.", while it would be fully functional on my laptop. Should probably sort this out before any further steps.

File: site/app/views/submission/HomeworkView.php [hunk 1]
```diff
@@ -240,9 +290,30 @@ public function showGradeable($gradeable, $days_late) {
...
+                                 {$svn_string},
+                                 {$gradeable->getNumTextBoxes()},
+                                 student_id);           
+        }
         $(document).ready(function() {
             $("#submit").click(function(e){ // Submit button
-                handleSubmission("{$this->core->buildUrl(array('component' => 'student',
+                var user_id = "";
+                if (document.getElementById("submissionForm"))
+                    user_id = document.getElementById("submissionForm").student_id.value;
```
Comment (MasterOdin) [inline site/app/views/submission/HomeworkView.php]: throw some braces around this.

**Comment by wangj35** [comment]
Thanks for the feedback @AndrewMRPI! I've made changes accordingly, but will wait until the team fix is merged into master before updating my branch on here.

**Comment by AndrewMRPI** [comment]
The user id is erased now when the radio buttons are switched to and from again but that could be a good thing.

**Pull request merged**